src/Google.Pubsub.V1/SimpleSubscription.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using static Google.Pubsub.V1.PublisherClient;
+using static Google.Pubsub.V1.SubscriberClient;
+
+
+namespace Google.Pubsub.V1
+{
+    public class SimpleSubscription
+    {
+        const int MaxMessages = 10;
+
+        public string ProjectId => Parent.ProjectId;
+        public string SubscriptionId { get; }
+        public string TopicId { get; }
+        public string TopicResourceName { get; }
+        public string ResourceName { get; }
+        public SimplePubsub Parent { get; }
+
+        public SimpleSubscription(SimplePubsub parent, string subscriptionId, string topicId)
+        {
+            Parent = parent;
+            SubscriptionId = subscriptionId;
+            TopicId = topicId;
+            ResourceName = GetSubscriptionName(parent.ProjectId, subscriptionId);
+            TopicResourceName = GetTopicName(parent.ProjectId, topicId);
+        }
+
+        // Options for completion:
+        // - CancellationToken
+        // - Return an IDisposable, and disposing of it stops (as per Rx)
+        // Other option: create another class which exposes an event. Or use IObservable.
+        // There are plenty of options for the async case, too...
+        // If we continue with the cancellation token, should we throw if the cancellation token is cancelled, or just return?
```

Comment:
Actually this could be called on a separate thread, if using the async method is not appropiate for whatever reason, so throwing is the expected way of dealing with cancelation then as it will unwrap the entire stack in the thread. Developers using cancelation token are expected to handle these errors and perform cleanup.